File: packages/jupyter-threeplot/src/scales/scale.ts
// Copyright (c) Jupyter Development Team.
// Distributed under the terms of the Modified BSD License.


import {
  WidgetModel, ManagerBase
} from '@jupyter-widgets/base';

import {
  JUPYTER_EXTENSION_VERSION
} from '../version';

import {
  BaseModel
} from '../base';



/**
 * TODO: Docstring
 */
export
abstract class ScaleModel extends BaseModel {
  defaults() {
    return {...super.defaults(),
      _model_name: ScaleModel.model_name,
    };
  }

  static serializers = {
      ...BaseModel.serializers,
      // Add any extra serializers here
    }

  static model_name = 'ScaleModel';
}

File: threeplot/scales/scale.py
#!/usr/bin/env python
# coding: utf-8

# Copyright (c) Jupyter Development Team.
# Distributed under the terms of the Modified BSD License.

"""
TODO: Add module docstring
"""

from ipywidgets import Widget
from traitlets import Unicode

from .._frontend import module_name, module_version


class ScaleWidget(Widget):
    """TODO: Add docstring here
    """
    _model_name = Unicode('ScaleModel').tag(sync=True)
    _model_module = Unicode(module_name).tag(sync=True)
    _model_module_version = Unicode(module_version).tag(sync=True)
